### ISSUE TITLE
Double scrollbars in grid

### ISSUE THREAD

**Issue by j71h** [ag-grid/ag-grid#7324]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Providing a Reproducible Scenario**
_Accepted reproducible scenarios are_
- _A description of the detailed steps to reproduce your behaviour in one of our examples in the docs._
- _A plunker_

_If you decide to send us a plunker, from any example in our website use the plunker button in there to fork your own code by following the steps below:_

- _Select the framework that is appropriate to you from the drop-down_
- _Open it in plunker. (Use the button plunker in our example)_
- _Add your changes so that the behaviour is reproduced_
- _Save and Freeze the plunker(On the top left corner)_
- _Send us the link to the plunker(You can copy the URL from the browser)_

If reporting a bug make sure to state.

Current behaviour.
Expected behaviour. If possible back this up with our docs/examples if possible

**Current behavior**

Double scrollbars are displayed
  
![image](https://github.com/ag-grid/ag-grid/assets/19264820/8776c055-5b52-4874-99a4-bd739633f7b1)

**Expected behavior**

Only single scrollbars is visible when the grid is larger then the available space.

**Please tell us about your environment:**

OS: Windows 11
IDE: Rider and MS Code
Package Manager: Vite
HTTP Server: Azure IIS

* **AG Grid version:** X.X.X

AG Grid version: 
        "ag-grid-community": "^30.2.1",
        "ag-grid-react": "^30.2.1",

* **Browser:**

Browser: Chrome 119.0.6045.200 (Official Build) (64-bit)

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

Javascript/Typescript

**Comment by kiril-matev** [comment]
Hello @j71h,

Thank you for this report - we're keen to reproduce and investigate this further.

Please note we understand you have a policy against sharing code, but all we need is the grid configuration, not any of your application code. 

I've built a simple example to try to reproduce this based on your screenshot - I can see you're using pinned columns and custom cell renderers. However, I can't reproduce this issue with this scenario:
https://plnkr.co/edit/YEfgkP3T6mRKNocP?open=main.js

Can you reproduce the issue with the plunker above? Please modify this plunker to reproduce this issue.

Let's try to narrow down this use case:
0/ In Windows Settings > Accessibility, at the top of the page you have a toggle **ALWAYS SHOW SCROLLBARS** - is this setting enabled?
![image](https://github.com/ag-grid/ag-grid/assets/60937283/d6e0149c-c2fc-4c6b-9006-a2508c0bebd8)
1/ Can you reproduce this issue if you don't have a pinned column in AG Grid?
2/ Can you reproduce this issue if you don't have custom cell renderers (try using default cell renderers)?
3/ In your gridOptions, are you setting alwaysShowHorizontalScroll, and alwaysShowVerticalScroll?
4/ When did you start seeing this double scrollbars issue:
4-a - After a Chrome upgrade without upgrading the AG Grid version?
4-b - Immediately after an AG Grid version update. Which AG Grid version did you upgrade from?

Looking forward to your response.

**Comment by j71h** [comment]
I can give quick responses to some of the questions.
0/ 
![image](https://github.com/ag-grid/ag-grid/assets/19264820/5331db70-826a-4a06-aeb5-a41de0acb2ae)
1/ The issue is still present even if no columns are pinned
2/ We are not using any `CustomRenderer` we use the `valueFormatter`
3/ Grid options does not include anything with regards to the scrollbar
     
![image](https://github.com/ag-grid/ag-grid/assets/19264820/c5e47d0c-ba84-4027-88b8-d820a6c083cc)
_**Image 1:** AG Grid instantiation_

![image](https://github.com/ag-grid/ag-grid/assets/19264820/d8b2980f-d5c4-4f8a-8ee9-50d4c9785dc1)
_**Image 2:** GridOptions_


4/ The double scrollbars appeared right after we upgraded from 25.3.0

**Comment by j71h** [comment]
As my old workaround had the unwanted side effect of disabling the scroll wheel I've tried to come up with a new one.
The vertical scrollbar was tagged with `<!--AG-FAKE-VERTICAL-SCROLL-->`, but I could not find the horizontal counterpart.

Anyhow this is my latest attempt.
``` css
div.ag-body>div.ag-body-vertical-scroll, div.ag-body-horizontal-scroll {
  display: none;
}
```

The removal of the vertical seems to be fine but the horizontal also removes the  vertical scrollbar for the pinned column.
![image](https://github.com/ag-grid/ag-grid/assets/19264820/8df8b0f1-cb4a-4f54-bb60-a6df2a19aca7)

**Comment by j71h** [comment]
0/ Sorry, I mistyped. the Windows version is Windows 10 Pro, 22H2, 19045.3693
5/ We are using ag-theme-alpine with an custom "ag-theme-se" to alter some colors, font and icons
    Applied like `<div className={"ag-theme-alpine ag-theme-se"}><AgGridReact ... /></div>`
6/ We do not change or set anything with regards to the scrollbars

Our custom styling
``` scss
@use "sass:map";

@mixin make-icon($name, $icon) {
  $url: map.get($icon, "url");
  @if map.has-key($icon, "rotate") {
    $rot: map.get($icon, "rotate");

    &.ag-icon-#{$name} {
      background-image: none;

      &:before {
        background-image: url($url);
        position: absolute;
        transform: rotate($rot);
        height: var(--se-icon-size);
        width: var(--se-icon-size);
      }
    }
  } @else {

    &.ag-icon-#{$name} {
      background-image: url($url);
    }

  }
}

.ag-theme-se {
  /* Override ag-grid variables to change some fundamental styling */
  --ag-font-family: Nunito, Roboto, Arial, Helvetica, sans-serif;

  --ag-header-foreground-color: #0f0f0f; /* se-black */
  --ag-header-background-color: var(--se-background-standard);
  --ag-even-row-background-color: var(--se-background-alternative);
  --ag-odd-row-background-color: #ecf7fc; /* se-sky-blue-10p */
  --ag-row-hover-color: var(--se-alternative-hover);
  --ag-selected-row-background-color: var(--se-alternative-selected);

  --ag-range-selection-border-color: var(--se-information);

  --ag-disabled-foreground-color: var(--se-disabled);

  --ag-header-height: 30px;
  --ag-grid-size: 4px;
  --ag-list-item-height: 25px;

  --se-icon-size: 16px;

  .ag-tooltip {
    background-color: var(--se-information);
    color: var(--se-information-contrast);
  }

  .ag-header-row {
    font-weight: 400;
  }

  .ag-popup-editor {
    background-color: var(--se-background-alternative);
    font-size: calc(var(--ag-font-size) + 1px);
  }

  /*https://github.com/ag-grid/ag-grid/issues/7324*/
  div.ag-body > div.ag-body-vertical-scroll, .ag-theme-se div.ag-body-horizontal-scroll {
    display: none;
  }

  .ag-checkbox {
    /* Simple override of checkbox styling to mimic the se-checkbox */
    .ag-input-wrapper {
      border: solid 1px var(--se-border);

      &:after {
        content: "" !important;
      }
    }

    .ag-checked {
      border: none;
      background-color: var(--se-success);

      &:after {
        background-image: url("data:image/svg+xml;...");  // svg data removed for brevity
        position: absolute;
        left: 50%;
        right: auto;
        top: 50%;
        bottom: auto;
        width: 16px;
        height: 12px;
        transform: translate(-50%, -50%) scale(0.8);
      }
    }

    .ag-indeterminate {
      border: none;
      background-color: var(--se-success);

      &:after {
        background-image: url("data:image/svg+xml;....");  // svg data removed for brevity
        position: absolute;
        left: 50%;
        right: auto;
        top: 50%;
        bottom: auto;
        width: 12px;
        height: 4px;
        transform: translate(-50%, -50%) scale(0.8);
      }
    }

  }

  .ag-icon {
    background-color: transparent;
    background-position: center;
    background-repeat: no-repeat;
    background-size: var(--se-icon-size) var(--se-icon-size);
    height: var(--se-icon-size);
    width: var(--se-icon-size);
    position: relative;

    &:before {
      content: "" !important;
    }
    
    $root-path: "@se/icons/svg";
    $icons: (
            "cancel": ("url": "#{$root-path}/action_delete_cross.svg"),
            "first": ("url": "#{$root-path}/arrow5_edge.svg", "rotate": "180deg"),
           // map shortened for brevity
    );

    @each $name, $icon in $icons {
      @include make-icon($name, $icon);
    }
  }
}
```

**Comment by kiril-matev** [comment]
5/ Please switch to using ag-theme-alpine only, without using ag-theme-se. 

Can you reproduce the scrollbars issue in this case?

**Comment by gportela85** [comment]
@j71h Looking at the screenshots provided, it looks like you are using scrollbars with custom styles (or at least it doesn't look like the default Windows scrollbar that I'm used to).

If this is the case, you need to make sure the scrollbars for the `ag-body-viewport` and `.ag-center-cols-viewport` are hidden, as they have been replaced by fake scrollbars within Ag Grid.

The style below should help:

```
.ag-body-viewport::-webkit-scrollbar, 
.ag-center-cols-viewport::-webkit-scrollbar {
    display: none !important;
}
```

If this is not the case, the comment above has a plunker that uses the theme provided in the issue description. As I was not able to see the issue there, can you help updating the example to show the double scrollbar issue?

**Comment by j71h** [comment]
@gportela85
> .ag-body-viewport::-webkit-scrollbar, 
> .ag-center-cols-viewport::-webkit-scrollbar {
>     display: none !important;
> }

The styling works and have no side effects that I can see right now. With my own attempts the "mouse" scrolling did not work in all directions.

I suspect that you might be on to something about the scrollbar styling, but I do not know from where that comes, need to investigate

**Comment by j71h** [comment]
Ok, looking into the applied styles I found this
![image](https://github.com/ag-grid/ag-grid/assets/19264820/02aeadfc-e891-44f6-a9cc-1ef3abf5b712)

Will apply it in the plunker to see if it is the cause

**Comment by kiril-matev** [comment]
Hello @j71h,

Thank you for your help with this - much appreciated!

Based on your description we've now reproduced this issue as a bug. AG Grid needs to prevent global styles from leaking inside AG Grid.

We have added this bug to our backlog and we are tracking it with the following reference and description:
**AG-10082 Global scrollbar styles leak into AG Grid scrollbars**

This bug will be fixed in our next release.

See whether this item will be part of our next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

**Now that this issue is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this bug is fixed in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/**

Thanks again for bringing this up with us - we appreciate your time!

**Comment by j71h** [comment]
@gportela85 
Yes, but from what I can see there was no `display` property applied in the code I provided, so I do not fully understand if what you have found is actually the problem I have...

**Comment by gportela85** [comment]
@j71h To know for sure what is causing the custom styles to force the scrollbar display within the grid, I would need access to your app so I could investigate the problem.
The example I provided above, with the `display: block` was one way to reproduce it, but I'm sure there could be more.

I can tell you that I have applied this as a permanent fix to our styles, so you shouldn't have a problem with future versions of Ag Grid. See: https://github.com/ag-grid/ag-grid/commit/1ddcca838ca0cb95e83022da45bdeb08054930fa

I'll close this issue for now, thanks for reporting the problem.
